index.js

```diff
@@ -1,81 +1,123 @@
+        currentIndex--;
+    
+        // And swap it with the current element.
+        [array[currentIndex], array[randomIndex]] = [array[randomIndex], array[currentIndex]];
+        }
+    
+        return array;
+    }
+    // Shuffling cards
+    shuffle(cardImageSrcs);
+
+    function checkForMatch(){
+        //if matched
+        if(flippedCards[0].children[0].src===flippedCards[1].children[0].src) // Checking if the flipped cards have same src i.e are matching
+        {        
+            matched++;
+            if(matched===6)
+            {
+                alert("hurrah! you did it")
+                //print the updated best score on the page
+                if(bestScore == null) 
+                    bestScore = counter;
+                else{
+                    bestScore = Math.min(bestScore, counter);
+                }
+                //store the best score on the local storage
+                localStorage.setItem("memory-game-best-score", bestScore);
+                bestTimer.innerHTML = "<b>" + bestScore + "</b>";
+                clearInterval(interval)
             }
-            //store the best score on the local storage
-            localStorage.setItem("memory-game-best-score", bestScore);
-            bestTimer.innerHTML = "<b>" + bestScore + "</b>";
-            clearInterval(interval)
+            else
+            alert("woah! matched")
+            
         }
+        //if not matched
         else
-        alert("woah! matched")
-        
-    }
-    //if not matched
-    else
-    {
-        flippedCards.forEach(flippedCard=>{
-            flippedCard.children[1].style.display="block";
-        })
-        alert("haha! better luck next time");
+        {
+            flippedCards.forEach(flippedCard=>{
+                flippedCard.children[0].src="#"; // Removing image src so that it isn't visible through HTML
+                flippedCard.children[0].alt="card front face"; // Removing image alt so that it isn't visible through HTML
+                flippedCard.children[1].style.display="block";
+            })
+            alert("haha! better luck next time");
+        }
+
+        flippedCards.length= 0;
     }
 
-    flippedCards.length= 0;
-}
+    function flipCard(e, i){
+        const card = e.target
 
-function flipCard(e){
-    const card = e.target
-    flippedCards.push(card)
-    card.children[1].style.display="none";
+        // This means that card is already flipped currently, so we exit out of function
+        if (card.children[0].src!==window.location.href+"#")
+            return;
 
-    //when we have filled two cards check for the match
-    if(flippedCards.length === 2)
-    {
-        setTimeout(checkForMatch,0); 
+        flippedCards.push(card)
+        card.children[0].src=cardImageSrcs[i]; // Setting image source for flipped front face
+        card.children[0].alt=cardImageSrcs[i].split('/').slice(-1)[0].slice(0, -4); // Setting image file name as alt text for flipped front face
```

Comment:
this works fine for '.png' but will not be accurate for other formats. Although doesn't matter much , but just for the sake of completeness.

Reply:
Done Sir, the code now removes the extension from the image file name and shows the rest as alt text